networkd/network_server.ml

```diff
@@ -314,52 +315,63 @@ module Interface = struct
             if Dhclient.is_running ~ipv6:true name then
               ignore (Dhclient.stop ~ipv6:true name);
             Sysctl.set_ipv6_autoconf name false;
-            Ip.flush_ip_addr ~ipv6:true name
+            Ip.flush_ip_addr ~ipv6:true name;
+            Sysctl.set_ipv6_enabled name false
           end
         | Linklocal6 ->
           if List.mem name (Sysfs.list ()) then begin
+            (* xapi sets Linklocal6 if IPv6 has not been configured;
+             * if IPv6 is not enabled globally, leave it disabled *)
             if Dhclient.is_running ~ipv6:true name then
               ignore (Dhclient.stop ~ipv6:true name);
-            Sysctl.set_ipv6_autoconf name false;
-            Ip.flush_ip_addr ~ipv6:true name;
-            Ip.set_ipv6_link_local_addr name
+            if Proc.get_ipv6_default_disabled () then
```

Comment:
I don't really get why we need to distinguish between "default" and "globally" disabled. Where is the thing that would actually set `/proc/sys/net/ipv6/conf/default/disable_ipv6` to `0`?

When will we actually end up in the "true" branch below, for "globally enabled" plus "default disabled"?